ref/vk/vk_devmem.c

```diff
@@ -66,138 +202,225 @@ static int allocateDeviceMemory(VkMemoryRequirements req, int type_index, VkMemo
+
+// NOTE(nilsoncore): I know, this is a mess... Sorry.
+// It could have been avoided by having short `VK_DevMemUsageTypes` enum names,
+// but I have done it this way because I want those enum names to be as descriptive as possible.
+// This basically replaces those enum names with ones provided by suffixes, which are just their endings. 
+#define REGISTER_STATS_METRICS( usage_type, usage_suffix ) { \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].allocations_current, STRING( allocations_current##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].allocations_current ), kSpeedsMetricCount ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].allocated_current, STRING( allocated_current##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].allocated_current ), kSpeedsMetricBytes ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].allocations_total, STRING( allocations_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].allocations_total ), kSpeedsMetricCount ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].allocated_total, STRING( allocated_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].allocated_total ), kSpeedsMetricBytes ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].frees_total, STRING( frees_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].frees_total ), kSpeedsMetricCount ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].freed_total, STRING( freed_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].freed_total ), kSpeedsMetricBytes ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].align_holes_current, STRING( align_holes_current##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].align_holes_current ), kSpeedsMetricCount ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].align_holes_size_current, STRING( align_holes_size_current##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].align_holes_size_current ), kSpeedsMetricBytes ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].align_holes_total, STRING( align_holes_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].align_holes_total ), kSpeedsMetricCount ); \
+ 	REGISTER_STATS_METRIC( g_vk_devmem.stats[usage_type].align_holes_size_total, STRING( align_holes_size_total##usage_suffix ), STRING( g_vk_devmem.stats[usage_suffix].align_holes_size_total ), kSpeedsMetricBytes ); \
 }
 
 qboolean VK_DevMemInit( void ) {
-	g_vk_devmem.verbose = !!gEngine.Sys_CheckParm("-vkdebugmem");
+	g_vk_devmem.verbose = gEngine.Sys_CheckParm( "-vkdebugmem" );
+
+	// Register standalone metrics.
+	R_SPEEDS_METRIC( g_vk_devmem.alloc_slots_count, "allocated_slots", kSpeedsMetricCount );
+	R_SPEEDS_METRIC( g_vk_devmem.device_allocated, "device_allocated", kSpeedsMetricBytes );
+	
+	// Register stats metrics for each usage type.
+	// Maybe these metrics should be enabled only with `-vkdebugmem`?
```

Comment:
No, they do make sense and are useful without. Their overhead is marginal.
```suggestion
```

Reply:
Removed.

https://github.com/nilsoncore/xash3d-fwgs/blob/79e0e520611965c4fde0cd7212d1164b7dd2cfce/ref/vk/vk_devmem.c#L379